### ISSUE TITLE
Add % of deviation when comparing two values and tolerance for this column is set to a percentage

### ISSUE THREAD

**Issue by aniketh02** [Seddryck/NBi#289]
When we apply tolerance - In the testresult it is giving the failure results (if the results are in tolerance range ) but how get the difference value between two numbers.

![non](https://user-images.githubusercontent.com/29744147/31270917-db71b0b6-aaa3-11e7-9093-5cca7e09d3c9.PNG)

**Comment by FuegoArtificial** [comment]
Could you please clarify what you are asking? Are you asking how to calculate a relative deviation?

=(new_value/original_value)-1

If I'm not mistaken it's +356% in your case.

For me personally, it's sufficient to have the result presented with its base values.

**Comment by aniketh02** [comment]
Thank you for the quick response. Yes, I am asking for relative difference in the result file. It is easy to know how much percentage difference is there b/w two numbers (if the tolerance in %).

**Comment by Seddryck** [comment]
It's really something easy to add but it adds really little value. If someone is motivated to do it, feel free, I'll merge this.

**Comment by riktimmondal** [comment]
I will like to take it

**Comment by Seddryck** [comment]
@riktimmondal great, feel free to ask questions if needed.

**Comment by aniketh02** [comment]
Thank you @riktimmondal. 
To add more context on the same, if we are comparing two numbers with no tolerance then the difference value should be displayed.

**Comment by Seddryck** [comment]
Could you explain why you need this? What are you doing with this information?

**Comment by FuegoArtificial** [comment]
My personal opinion: I like the base values to track down the reason easier. That a deviation is out of bounds tells the error message already.
However, it's a minor thing, I won't argue about it :)

**Comment by Seddryck** [comment]
@FuegoArtificial I also make use of base values to understand the root cause of the red test. On the other hand, I don't see a case where it would have been easier to have a % (or a difference) to have a better understanding of the issue. I'm really curious to have a case where it helps.

**Comment by aniketh02** [comment]
The Case:- Because of changes to the code, We are aware the expected  difference is  a few hundred thousands here the difference seen is -26,454,611 and to Highlight the percentage of  deviation, this helps Prioritize the Fix in  a release.

So displaying the difference or Percentage helps the developers and testers flag the importance and rank the issues by the percentage of deviation.

**Comment by aniketh02** [comment]
@riktimmondal could you please update on the status of this request.

**Comment by FuegoArtificial** [comment]
@aniketh02 I see your point. However, I believe, that a test framework is not the right tool to figure priorities. The test is a pass or no pass decision.
If it fails, a process of your choice begins where people decide on whichever reasons how to handle the failed test.
Your way is a very specific one to handle a failed test (a number of prioritized tasks). For the next failed test, the way could be different and the deviation would be clutter information for you, irrelevant and harder to find the reason of the failed test for you.
Each case how to specifically handle a failed test can be different. With your test you find the problems, what the test framework is for. 
Perhaps, it could be some sort of property of the test definition how to display the error message? So it optionally shows percentage instead of the base values? :-)

**Comment by aniketh02** [comment]
Thanks Fuego,
 I think, I mixed both of them (Failure's and Prioritization), let me take a step back on Prioritization and detail on Failure.
1. When a developer is Unit testing and he knows that there is an issue, It a Qualitative fact, that it is a pass or a fail
2. If the developer knows the size of the issue, that is quantitative, he can justify that the code is not causing a great harm, or causing a Big Harm to the numbers, by providing the difference of values and percentages, it helps quantify an issue in his code.
The Idea "Perhaps, it could be some sort of property of the test definition how to display the error message? So it optionally shows percentage instead of the base values? " is great

**Comment by Seddryck** [comment]
@aniketh02 I agree with your point 1 but not the 2. As @FuegoArtificial explained, you've many ways to define the priorities: it could be the count of rows with a failure, the delta between actuals and expectations, the percentage, the variance of the deltas, ... Too many options to rely on a test framework to perform this analysis for you.

IMO, if you want to automatically sets priorities on your bugs (or even manually review the priorities assisted by a software creating a report) then you should rely on a tool processing your test-results and not on the test framework. That's just separation of concerns.

Post-processing of test-results can be facilitated by #307, there you'll have the option to setup a second tool (not aware of any tool doing such a thing) parsing the test-results and applying your own calculations, enriching the results and facilitating your own prioritizations. If you want to specify in the test what kind of calculations or prioritizations should be applied by this other tool, feel free to use the ```traits``` ... that's what they are intended to do: specify parameters that should be used by other tools.

The more we're speaking about it, the less I think it's something for NBi. I'm thinking to close this issue by ```out-of-scope```, if we've no new arguments in favor to do such things without impacting all users.

@riktimmondal have you already started to work on this?
